lib/liquid/ruby_tokenizer.rb

```diff
@@ -0,0 +1,28 @@
+module Liquid
+  class RubyTokenizer
```

Comment:
Agreed.  We should just have an interface similar to liquid-c, where it passes a Tokenizer object rather than an array of tokens to Document.parse, and Tokenizer just implements the `shift` method, which you can delegate to `@tokens.shift` in the ruby implementation.

From liquid-c it might be easier to define LiquidC::Tokenizer to delegate to until it exposes line numbers.